### PR TITLE
Get email from contact book when a name is entered

### LINKED ISSUE #37: Get reciever from name
If a user enters a name (and not an email) in the NLP the automation should be able to find the intended receiver in a contact book or something similar

Example if the user enters: "send an email to Per that I want to meet tomorrow"  then the automation should look up the contact and find the email to "Per".

The lookup could perhaps be made via the email api or through a local contact book.
If multiple contacts with the same name exists then prompt the user on who is the correct one

### PR DESCRIPTION
**Pull request by 97gushan**
If a user enters the receiver of an email this string might or might not be an actual email address. 
This string is checked against a regex to see if it is an email address. If it is not then possible receivers are returned. If there are multiple users with the same or similar names the user is prompted and able to enter the correct one. 
When the intended receiver is found the email is sent. 

Can be tested by running `email_name_demo.py`, this test does not use the NLP and the inputted name is set to `John`.

Closes #37

### CODE REVIEW

File: lib/automate/modules/send.py [hunk 1]
```diff
@@ -14,26 +16,55 @@ def __init__(self):
...
+            receiver = to[0]
+        elif len(to) == 0:
+            raise NoReceiverError("No receiver was entered")
+        else:
+            raise ToManyReceiversError("Can only handle one (1) receiver at this time")
+
+        if not self.is_email(receiver):
+            # filter out the contacts that does not need to be considered
+            possible_receivers = fuzzy.extract(receiver, SETTINGS["users"].keys())
+            possible_receivers = list(filter(lambda x: x[1] > 75, possible_receivers))
```
Comment (widforss) [inline lib/automate/modules/send.py:35]: You should probably remove the sender here as well, you rarely want to send an email to yourself (I only do it for primitive file sharing between computers, and that probably shouldn't be considered.).

File: lib/automate/modules/send.py [hunk 2]
```diff
@@ -14,26 +16,55 @@ def __init__(self):
...
+
+            # if there are multiple possible receivers then return a string of these that will
+            # be displayed to the user.
+            #
+            # The user will then be able to enter a more precise name
+            # that will be sent to the follow-up method
+            if len(possible_receivers) > 1:
+                names = "\n".join(list(map(lambda contact: contact[0], possible_receivers)))
+                return None, "Found multiple contacts: \n" + names + "\nPlease enter the name"
```
Comment (widforss) [inline lib/automate/modules/send.py:44]: We will probably need an attribute in the future to keep track of *what* follow up question we are asking. But as you are the first one to use this functionality this isn't needed here.

**Review by widforss — APPROVED** [review]
Good job! I'd like to work on an issue building on this, integrating it with an external contact book. Is it ok for you or are you planning on doing that?

File: lib/automate/modules/send.py (hunk 1 above)
Comment (97gushan) [reply lib/automate/modules/send.py:35]: I can filter it out for now, we should probably split the local contact book and the information about the user but this is solved when using an external contact book. You can go ahead and look into the external if you want to

**Review by banunkers — APPROVED** [review]
Sweet :+1:

**Pull request merged**